pkg/alertmanager/asset_store.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alertmanager
+
+import (
+	"context"
+	"fmt"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
```

Comment:
Do we really need to duplicate `pkg/prometheus/asset_store` here?

Reply:
For this I was using the [_Rule of Three_ for refactoring](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)), probably because I didn't want to have to think about where best to move it to. If you want me to de-duplicate, just let me know where in the project I should move the existing asset_store code to. :slightly_smiling_face: 

Alternatively, the de-duplication could be done in a separate change set (this duplication doesn't really expose any more functions or types publicly from the project, but de-duplication would require some things to be exposed).

Reply:
You were right with doing so, but I can definitely see a third duplication occurring overtime.
Since refactoring would require to rewrite the original code partially, I think we can keep it as is here and do the refactor as part of another effort.

From the existing architecture, I'd say that the `k8sutil` package is the most neutral place to host this code.

cc @simonpasquier as you recently wrote this package and might have some thoughts on that.